microsoft-edge/devtools-guide-chromium/memory-problems/memory-101.md

```diff
@@ -25,155 +25,182 @@ keywords: microsoft edge, web development, f12 tools, devtools
+Distance from root:
 
-:::image type="complex" source="../media/memory-problems-root.msft.png" alt-text="Distance from root" lightbox="../media/memory-problems-root.msft.png":::
-   Distance from root
-:::image-end:::
+:::image type="content" source="../media/memory-problems-root.msft.png" alt-text="Distance from root." lightbox="../media/memory-problems-root.msft.png":::
 
 
 <!-- ====================================================================== -->
 ## Dominators
 
-Dominator objects are comprised of a tree structure because each object has exactly one dominator.  A dominator of an object may lack direct references to an object it dominates; that is, the tree of the dominator is not a spanning tree of the graph.
+Dominator objects are comprised of a tree structure, because each object has exactly one dominator.  A dominator of an object might lack direct references to an object that it dominates.  That is, dominator's tree is not a spanning tree of the graph.
 
-In the following figure, the following statement are true.
+In the following figure:
 
-*   Node 1 dominates node 2
-*   Node 2 dominates nodes 3, 4 and 6
-*   Node 3 dominates node 5
-*   Node 5 dominates node 8
-*   Node 6 dominates node 7
+*   Node 1 dominates node 2.
+*   Node 2 dominates nodes 3, 4 and 6.
+*   Node 3 dominates node 5.
+*   Node 5 dominates node 8.
+*   Node 6 dominates node 7.
 
-:::image type="complex" source="../media/memory-problems-dominatorsspanning.msft.png" alt-text="Dominator tree structure" lightbox="../media/memory-problems-dominatorsspanning.msft.png":::
-   Dominator tree structure
-:::image-end:::
+:::image type="content" source="../media/memory-problems-dominatorsspanning.msft.png" alt-text="Dominator tree structure." lightbox="../media/memory-problems-dominatorsspanning.msft.png":::
 
-In the following figure, node `#3` is the dominator of `#10`, but `#7` also exists in every simple path from Garbage Collector to `#10`.  Therefore, an object B is a dominator of an object A if B exists in every simple path from the root to the object A.
+In the following figure, node `#3` is the dominator of node `#10`.  But node `#7` also exists in every simple path from the garbage collection root **GC** to node `#10`. Therefore, an object `B` is a dominator of an object `A` if object `B` exists in every simple path from the root to the object `A`.
 
-:::image type="complex" source="../media/memory-problems-dominators.msft.gif" alt-text="Animated dominator illustration" lightbox="../media/memory-problems-dominators.msft.gif":::
-   Animated dominator illustration
-:::image-end:::
+Node `GC` dominates nodes `#1`, `#3`, and `#11`:
+
+![Node GC dominates nodes #1, #3, and #11.](../media/memory-problems-dominators-00.msft.png)
```

Comment:
In some of the images 00-05, the main point is obscured because the mouse pointer is in the way of the number in the middle of the circle.  The original looping gif in some cases, has the same issue: https://docs.microsoft.com/en-us/microsoft-edge/devtools-guide-chromium/memory-problems/memory-101#dominators

Reply:
Will fix.

Reply:
SME approved as-is